tests/integration/cocoon/test_reference_portfolios.py

```diff
@@ -0,0 +1,171 @@
+import unittest
+from pathlib import Path
+import pandas as pd
+import lusid
+from lusidtools import cocoon as cocoon
+from lusidtools import logger
+from lusidtools.cocoon.utilities import create_scope_id
+from parameterized import parameterized
+
+
+class CocoonTestsReferencePortfolios(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+        cls.portfolios_api = cls.api_factory.build(lusid.api.PortfoliosApi)
+        cls.unique_id = create_scope_id()
+        cls.logger = logger.LusidLogger("debug")
+        cls.scope = "ModelFundTest"
+        cls.file_name = "data/reference-portfolio/reference-test.csv"
+
+    @parameterized.expand(
+        [
+            [
+                "Load a reference portfolio with required attributes only",
+                "data/reference-portfolio/reference-test.csv",
+                {"code": "FundCode", "display_name": "display_name"},
+                {
+                    "description": "description",
+                    "created": "created",
+                    "base_currency": "base_currency",
+                },
+                [],
+            ],
+            [
+                "Load a reference portfolio with required attributes AND properties",
+                "data/reference-portfolio/reference-test.csv",
+                {"code": "FundCode", "display_name": "display_name"},
+                {
+                    "description": "description",
+                    "created": "created",
+                    "base_currency": "base_currency",
+                },
+                ["strategy", "custodian"],
+            ],
+            [
+                "Load duplicate reference portfolios",
+                "data/reference-portfolio/reference-test-duplicates.csv",
+                {"code": "FundCode", "display_name": "display_name"},
+                {
+                    "description": "description",
+                    "created": "created",
+                    "base_currency": "base_currency",
+                },
+                [],
+            ],
+        ]
+    )
+    def test_load_from_data_frame_a_portfolios_failure(
```

Comment:
This should be renamed to indicate it is looking for successes, not failures.

Reply:
Agreed. Good catch. Updated.